packages/grafana-ui/src/utils/statsCalculator.ts

```diff
@@ -0,0 +1,402 @@
+// Libraries
+import isNumber from 'lodash/isNumber';
+
+import { TableData, NullValueMode } from '../types/index';
+
+export enum StatID {
+  sum = 'sum',
+  max = 'max',
+  min = 'min',
+  logmin = 'logmin',
+  mean = 'mean',
+  last = 'last',
+  first = 'first',
+  count = 'count',
+  range = 'range',
+  diff = 'diff',
+  delta = 'delta',
+  step = 'step',
+
+  changeCount = 'changeCount',
+  distinctCount = 'distinctCount',
+
+  allIsZero = 'allIsZero',
+  allIsNull = 'allIsNull',
+}
+
+export interface ColumnStats {
```

Comment:
type ColumnStats = {
    [P in keyof StatID]?: number;
}
might have to be number |  boolean to support allIsZero etc

Reply:
should be any, since table data can be any.

first/last string is *super* common.  

The `StatID` enum is there to make getting the standard ones easy, but I don't think it needs to be the list of everything.

I originally called this `TableReducer` since that is really what it does... but went back to 'stats' so the mental lift was less extreme.